### PR TITLE
HOTT-1867 - Redirect search at the frontend according to meta data

### PR DESCRIPTION
**Pull request by javier-transform**
### Jira link

[HOTT-1867](https://transformuk.atlassian.net/browse/HOTT-1867)

### What?

I have added/removed/altered:

- [x] Redirect search at the frontend according to meta data

### CODE REVIEW

File: app/services/beta/search/perform_search_service.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ def call
         path = "/api/beta/search?#{query_params}"
         response = api.get(path)
         parsed = parse_jsonapi(response)
+        merge_meta_to_result!(parsed, response)
```
Comment (willfish) [inline app/services/beta/search/perform_search_service.rb]: This shouldn't be necessary AFAIK

Comment (javier-transform) [reply app/services/beta/search/perform_search_service.rb]: When the parse_jsonapi returns the object there is no meta merged into it. There is a statement to check and merge meta inside the tariff parser api but the meta data does not exist in the `data` part of the response. That is why we need to merge the meta part and i added this line.
image

Comment (javier-transform) [reply app/services/beta/search/perform_search_service.rb]: I didn't remove the code inside tariff parser api or alter it since there could be edge case(s) regarding that, and that's why i waited for your review.

Comment (willfish) [reply app/services/beta/search/perform_search_service.rb]: Oh, okay. Probably we should add a parse_top_level_meta! method to api entity and add tests for that. I don't think it should affect any of the existing api entity models negatively/have any crazy edge cases

**Review by willfish — APPROVED** [review]
